devito/ir/support/stencil.py

```diff
@@ -106,6 +106,17 @@ def union(cls, *dicts):
                 output[k] |= v
         return output
 
+    @classmethod
+    def retrieve_offsets(cls, stencils):
+        """
+        Return a mapper from :class:`Dimension`s to the min/max integer offsets
+        within ``stencils``.
+        """
+        offs = Stencil.union(*stencils)
+        mapper = {d: v for d, v in offs.diameter.items()}
+        mapper.update({d.parent: v for d, v in mapper.items() if d.is_Stepping})
```

Comment:
oh, no! this is too bad! 
``d.is_Stepping`` is a concept that shouldn't live here. It's completely unrelated.
Unfortunately, I only noticed it now: I suggest we move this whole routine to the only place where it's used.

Reply:
Fixed